recipes/cofi-espresso/meta.yaml

```diff
@@ -0,0 +1,53 @@
+{% set name = "cofi-espresso" %}
+{% set version = "0.0.2.dev0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/cofi_espresso-{{ version }}.tar.gz
+  sha256: c874b36076ea646a3f9d19e8896198a28c8a1db28ad704c7d120c044963d1fa6
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  missing_dso_whitelist:
+    - $RPATH/libgfortran.so.5
+    - /lib64/libm.so.6
+    - /lib64/libc.so.6
+    - lib/libgcc_s.so.1
+    - lib/libquadmath.so.0
```

Comment:
This is shipping "fake" pre-compiled object files. The project has an ad-hoc build system - for some reason, we decide to delay compilation until a specific piece of compiled code is called.

In short:
1. We ship some C / Fortran source code as data files
2. We execute an installation script when a piece of C / Fortran code is needed
3. This will lead to new files being generated in site-packages
4. When users uninstall this package, `pip` won't remove those files generated after the new files
5. I want the package to be cleaned thoroughly when users run `pip uninstall`, so added empty `*.o` / `*.so` files to ship with the package
6. These files are empty, and they are not pre-compiled files at all, but then led to conda-forge infrastructure to complain about missing dependencies, so I added these lines to `meta.yaml`

Thanks for reviewing this carefully. Please let me know if there's an issue with this handling.

Reply:
Ideally 1 and 2 are executed at build time and packaged via the usual means, so there's less magic going on the user end. Is there anything preventing pre-compilation?

Reply:
Yes, say we have the following structure:
```
package
|__ submodule 1
|__ submodule 2
|__ submodule 3
```

In the use case of this specific package, the submodules are independent of one another. *We don't want the failure of building in `submodule 1` to affect the usage of `submodule 3`.* If it fails to install in the first place, then users won't be able to use any of the other submodules. The nature (and resources) of this project doesn't guarantee that every submodule will be polished and fully tested on every possible type of machines and OS. That's why we want to delay the build process to prevent bad user experience.

I understand it doesn't make sense from a developer's point of view (since it's not what's usually done in a Python package?). However the users are mostly going to be scientific researchers / students who don't have much experience in compiling, linking, fixing Python installation stuff (or even virtual environments). I hope it makes more sense now?

Reply:
Once (built and) packaged, it will install reliably. The "only" thing `conda` does is extract an archive and link it into the target environment (big generalization, but the point remains).

The only hiccup you might get is updating the package on the feedstock for a new release when "submodule 1" is not polished or something, but that's a burden on you / the maintainers, not the end user.

Does that make sense? I am basically suggesting you build everything here in conda-forge, providing a package that has everything precompiled already.